Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+TypeSignatureDisambiguation.swift

```diff
@@ -0,0 +1,437 @@
+        })
+        
+        guard !typeNameIndicesToCheck.isEmpty else {
+            // Every type name is common across all overloads. These type names can't be used to disambiguate the overloads.
+            return .init(repeating: nil, count: typeNames.size.width)
+        }
+        
+        // Create a sequence of type name combinations with increasing number of type names in each combination.
+        let typeNameCombinationsToCheck = typeNameIndicesToCheck.combinationsToCheck()
+        
+        return typeNames.rowIndices.map { row in
+            var shortestDisambiguationSoFar: (indicesToInclude: IntSet, length: Int)?
+            
+            for typeNamesToInclude in typeNameCombinationsToCheck {
+                // Stop if we've already found a disambiguating combination using fewer type names than this.
+                guard typeNamesToInclude.count <= (shortestDisambiguationSoFar?.indicesToInclude.count ?? .max) else {
+                    break
+                }
+                
+                // Compute which other overloads this combinations of type names also could refer to.
+                var iterator = typeNamesToInclude.makeIterator()
+                let firstTypeNameToInclude = iterator.next()! // The generated `typeNamesToInclude` is never empty.
+                let overlap = IteratorSequence(iterator).reduce(into: table[row, firstTypeNameToInclude]) { accumulatedOverlap, index in
+                    accumulatedOverlap.formIntersection(table[row, index])
+                }
+                
+                guard overlap.count == 1 else {
+                    // This combination of parameters doesn't disambiguate the result
+                    continue
+                }
+                
+                // Track the combined length of this combination of type names in case another combination (with the same number of type names) is shorter.
+                let length = typeNamesToInclude.reduce(0) { accumulatedLength, index in
+                    // It's faster to check the number of UTF8 code units.
+                    // This disfavors non-UTF8 type names, but those could be harder to read/write so neither length is right or wrong here.
+                    accumulatedLength + typeNames[row, index].utf8.count
+                }
+                if length < (shortestDisambiguationSoFar?.length ?? .max) {
+                    shortestDisambiguationSoFar = (IntSet(typeNamesToInclude), length)
+                }
+            }
+            
+            guard let (indicesToInclude, _) = shortestDisambiguationSoFar else {
+                // This overload can't be uniquely disambiguated by these type names
+                return nil
+            }
+            
+            // Found the fewest (and shortest) type names that uniquely disambiguate this overload.
+            // Return the list of disambiguating type names or "_" for an unused type name.
+            return typeNames.columnIndices.map {
```

Comment:
Is it fewest or shortest? Reading the code above (the `reduce(0)` snippet that calculates the length) it seems like this algorithm returns the overall shortest string, and not necessarily the string which contains the fewest number of disambiguating types. Is this correct?

Maybe refine the comment here to explain.